src/views/W3NRemove/W3NRemove.tsx

```diff
@@ -0,0 +1,207 @@
+import { useCallback } from 'react';
+import { Link, generatePath } from 'react-router-dom';
+import { browser } from 'webextension-polyfill-ts';
+import { Deposit, IDidIdentifier } from '@kiltprotocol/types';
+
+import { FullDidDetails, Web3Names } from '@kiltprotocol/did';
+
+import { BlockchainApiConnection } from '@kiltprotocol/chain-helpers';
+
+import { Option, Struct } from '@polkadot/types';
+import { AnyNumber } from '@polkadot/types/types';
+
+import * as styles from './W3NRemove.module.css';
+
+import { Identity } from '../../utilities/identities/types';
+
+import {
+  getKeypairBySeed,
+  getKeystoreFromSeed,
+} from '../../utilities/identities/identities';
+import { paths } from '../paths';
+
+import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
+import {
+  PasswordField,
+  usePasswordField,
+} from '../../components/PasswordField/PasswordField';
+import { KiltCurrency } from '../../components/KiltCurrency/KiltCurrency';
+import {
+  asKiltCoins,
+  KiltAmount,
+} from '../../components/KiltAmount/KiltAmount';
+import { TxStatusModal } from '../../components/TxStatusModal/TxStatusModal';
+import { LinkBack } from '../../components/LinkBack/LinkBack';
+import { Stats } from '../../components/Stats/Stats';
+
+import { getFullDidDetails } from '../../utilities/did/did';
+
+import { getExtrinsicFee } from '../../utilities/getExtrinsicFee/getExtrinsicFee';
+import { useSwrDataOrThrow } from '../../utilities/useSwrDataOrThrow/useSwrDataOrThrow';
+
+import { useSubmitStates } from '../../utilities/useSubmitStates/useSubmitStates';
+
+async function getFee(fullDid?: FullDidDetails) {
+  if (!fullDid) {
+    return undefined;
+  }
+
+  const fakeSeed = new Uint8Array(32);
+  const keypair = getKeypairBySeed(fakeSeed);
+
+  const extrinsic = await Web3Names.getReleaseByOwnerTx();
+
+  const authorized = await fullDid.authorizeExtrinsic(
+    extrinsic,
+    await getKeystoreFromSeed(fakeSeed),
+    keypair.address,
+  );
+  const blockchain = await BlockchainApiConnection.getConnectionOrConnect();
+  const signed = await blockchain.signTx(keypair, authorized);
+  return getExtrinsicFee(signed);
+}
+
+interface Web3NameData extends Struct {
+  owner: IDidIdentifier;
+  claimedAt: AnyNumber;
+  deposit: Deposit;
+}
+
+async function getDepositData(did: string) {
+  const web3name = await Web3Names.queryWeb3NameForDid(did);
+
+  const { api } = await BlockchainApiConnection.getConnectionOrConnect();
+
+  const { deposit } = (
+    await api.query.web3Names.owner<Option<Web3NameData>>(web3name)
+  ).unwrap();
```

Comment:
not that unwrap throws if the did web3name is not registered

Reply:
That error would be caught, but probably an edge case since the user would not be able to reach this removal page if the web3name is not found on the blockchain.